core/javatests/com/google/instrumentation/stats/AggregationDescriptorTest.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2016, Google Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.instrumentation.stats;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.instrumentation.common.Duration;
+import com.google.instrumentation.common.Function;
+import com.google.instrumentation.stats.AggregationDescriptor.DistributionAggregationDescriptor;
+import com.google.instrumentation.stats.AggregationDescriptor.IntervalAggregationDescriptor;
+import java.util.Arrays;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/**
+ * Tests for {@link AggregationDescriptor}
+ */
+@RunWith(JUnit4.class)
+public final class AggregationDescriptorTest {
```

Comment:
For these tests, I just want to verify that the data is wired up as expected.

Once the descriptors and associated data classes are checked in, I plan to add some examples that I hope will demonstrate how everything works together.